src/core/interpreter.cc

```diff
@@ -993,7 +969,11 @@ int Interpreter::RedisGenericCommand(bool raise_error, bool async, ObjectExplore
   /* Pop all arguments from the stack, we do not need them anymore
    * and this way we guaranty we will have room on the stack for the result. */
   lua_pop(lua_, argc);
+  return std::make_optional(args);
+}
 
+std::optional<int> Interpreter::CallRedisFunction(bool* raise_error, bool async,
```

Comment:
I think the interface here is too much. It has optional that wraps int that based on the code can return only 0 and 1, and we have raise_error that is another output argument?
1. CallRedisFunction should return number of arguments it put on stack. Does CallRedisFunction ever return 0? Is lua stack really on 0 when it happens  ?  
2. raise_error seems to be relevant only if CallRedisFunction returns nullopt.

Maybe introduce `CallResult = variant<int, bool>` so if the first argument is defined then it's number of results and the second is raise_error?

Reply:
simplified the flow